halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/ui/v1/Ui.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.ui.v1;
+
+/**
+ * A shared colored terminal output class
+ */
+public class Ui {
+  private Color color;
+  private boolean verbose;
+
+  public Ui(boolean colorEnabled, boolean verbose) {
+    this.color = new Color(colorEnabled);
+    this.verbose = verbose;
+  }
+
+  private void reset(String message) {
+    try {
+      System.out.print(message);
+    } catch (Exception e) {
+      throw e;
+    } finally {
+      System.out.println(color.RESET);
+    }
+  }
+
+  public void raw(String message) {
+    reset(message);
+  }
+
+  public void info(String message) {
+    if (verbose) {
+      reset(color.BOLD + color.BLUE + ". " + color.RESET + message);
+    }
+  }
+
+  public void warning(String message) {
```

Comment:
This becomes output(
    [new PrintablePhrase(Flavor.combine(color.BOLD, color.YELLOW), "-"),
     new PrintablePhrase(Color.NONE, message)]
)
This is ugly since java doesnt have operator overloading (which I dont normally like anyway).  Since the first phrase is constant, you can precompute that into a constant PrintablePhrase.

Here it looks tedious, but it means that you could have multicolored text within the message itself since the output takes a printable phrase.

reset() becomes something like outputAtLevel(Level, Iterable<PrintablePhrase>)
warning, etc are these types of shorthand as you have here, but you could have a warning(PrintablePhrase) which would permit multicolored text which could be useful for more complicated output scenarios.

 The Level could be defined like your colors
WARNING = new PrintablePhrase(Flavor.combine(Weight.BOLD, Color.YELLOW), "- ")

Given a constructor PrintablePhrase(String message) : PrintablePhrase(Flavor.NONE, message)

Then warning(message) becomes
outputAtLevel(WARNING, PrintablePhrase(message))

outputAtLevel(Level level, PrintablePhrase phrase)  becomes
output([level, phrase])
where output adds the RESET when sequencing across phrases.

Still tedious at the lowest levels but looks concise and expressive for the use sites.